### PR TITLE
Delvelop column limit2

### PR DESCRIPTION
**Pull request by BerndGit**
Now I have implemented a soft limit for Cards per column/row. 

To be accessed via Context Menu in the headers ("Edit Properties").

To save the data, I had to extend IDim by:
        bool LimitSet { get; set; }
        int MaxNumberOfCards { get; set; }

Backwards compatible to old databases.

### CODE REVIEW

**Comment by Liklainy** [comment]
@BerndGit, @Totopolis may be the card limit per intersection will be more relevant and needed, than the limits per column/row?

**Comment by BerndGit** [comment]
@Liklainy :   I use the column limit to implement an Work-in-Progress (WIP) Limit  (see e.g. https://kanbanzone.com/kanban-resources/kanban-wip-limits/)

Rows are my projects.
Columns include: Backlog | This Month | this week | Today | Now |  Waiting for others  |Done

If i want to set a max limit of cards for today, for me the sum over all projects is more relevant than for an individual project.

Therefore I have implemented a Column Limit. The Row limit is just there for symentry and since it was easy to add. 

I don't know a use case for an intersection limit. But if you see a potential application for it we could implement it as a limit additionally to the row/column limit.

File: KambanSolution/Kamban.Common/Kamban.Common.csproj [hunk 1]
```diff
@@ -1,9 +1,13 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net462</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="DynamicData" Version="6.12.13" />
```
[the remark this answers is not in the document]
Comment (BerndGit) [reply KambanSolution/Kamban.Common/Kamban.Common.csproj]: I assume there is no need. It's just an artifact from previous tries.  I'll change to version of Topolis:master

File: KambanSolution/Kamban/Kamban.csproj [hunk 2]
```diff
@@ -34,6 +34,10 @@
     <Compile Update="Properties\Resources.Designer.cs" DesignTime="True" AutoGen="True" DependentUpon="Resources.resx" />
 
     <None Update="Properties\Settings.settings" Generator="SettingsSingleFileGenerator" LastGenOutput="Settings.Designer.cs" />
+
+    <None Update="Views\HeaderPropertyView.xaml">
```
[the remark this answers is not in the document]
Comment (BerndGit) [reply KambanSolution/Kamban/Kamban.csproj]: Removed

**Comment by BerndGit** [comment]
Done now

**Comment by BerndGit** [comment]
Changes applied and conflicts resolved

File: KambanSolution/Kamban/Views/HeaderPropertyView.xaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        
+        <Label Grid.Row="2" Grid.Column="1" Content="Name" HorizontalAlignment="Right" VerticalAlignment="Center" BorderThickness="0"/>
+        <TextBox x:Name="TB_HeaderName"   Grid.Row="2" Grid.Column="2" Text ="{Binding Path=HeaderName}" Width="150" HorizontalAlignment="Left" VerticalAlignment="Center"/>
+
+        <Label Grid.Row="3" Grid.Column="1" Content="Card Limit " HorizontalAlignment="Right" VerticalAlignment="Center" BorderThickness="0"/>
+        <CheckBox Grid.Row="3" Grid.Column="2" x:Name="CB_CardLimit" HorizontalAlignment="Left" VerticalAlignment="Center" IsChecked="{Binding Path=HeaderLimitSet}"/>
+
+        <Label Grid.Row="4" Grid.Column="1" Content="Max Cards" HorizontalAlignment="Right" IsEnabled="{Binding ElementName=CB_CardLimit, Path=IsChecked}" VerticalAlignment="Center" BorderThickness="0"/>
+        <TextBox Grid.Row="4" Grid.Column="2" Width="80" Text ="{Binding Path=HeaderMaxNumber}" IsEnabled="{Binding ElementName=CB_CardLimit, Path=IsChecked}"  PreviewTextInput="NumberValidationTextBox" HorizontalAlignment="Left" VerticalAlignment="Center"/>
```
[the remark this answers is not in the document]
Comment (BerndGit) [reply KambanSolution/Kamban/Views/HeaderPropertyView.xaml]: Thanks for that hint

**Comment by BerndGit** [comment]
> @BerndGit, Thank you for your hard work!
> The PR looks good for me, I've only got a few minor comments.

Thank you, for your review @Liklainy 

I like the tool Kamban. And I really use it in my work. It was not so easy to find a functional, responsive tool, which does not store data in the cloud. (My company would not allow cloud based tools.)

Since I like to have some more functionality I started to help myself. Actually I never was working with c# or wpf before. I have some Java experience but I'm not a professional programmer. Basicall I learnt a lot, when I was just analysing the existing code. 
.. so if i do some strange things here feel free to correct me. 

Some more pull requests, with features I like might come ...

**Pull request merged**